src/neural/factory.h

```diff
@@ -80,6 +93,11 @@ class NetworkFactory {
 
   std::vector<Factory> factories_;
   friend class Register;
+
+  // Cached values.
+  static std::string network_path_;
```

Comment:
Why is this static? What happens in selfplay when player1 and player2 use different network files and/or backends?

Reply:
Maybe store `BackendConfiguration network_configuration_` (see comment in tournament.h what it is) in `EngineController` (in addition to network_), and then load new network if `network_configuration_ != BackendConfiguration(options)`.
E.g. let calling code watch what is loaded and when there's need to reload network, rather than do that inside factory.